piper/base/executors/clip.py

```diff
@@ -0,0 +1,39 @@
+from fastapi import FastAPIExecutor
+from piper import envs
+
+import torch
+import requests
+import clip
+import PIL
+
+class ClipExecutor(FastAPIExecutor): 
+
+    def __init__(self, url:str, text_snippets:list[str]):
+            self.url = url
+            self.text_snippets = text_snippets
+            self.model = "ViT-B/32"
+
+    def download_picture(self) -> PIL.Image.Image:
+        return PIL.Image.open(requests.get(self.url, stream=True).raw)
+
+    def do_clip(self) -> list:
```

Comment:
Please, change  the naming or add async "run" method (it's async right now, would be sync in the future)

async def run(self) instead do_clip

It would automatically calls this implementation in HTTPExecutor via FatAPIExecutor when you call __call__ in CLIPExecutor

https://github.com/TatraDev/pipertool/blob/27d9a5083890c3073e38aec006789fcc722c4549/piper/base/executors/http.py#L39